lib/nx_zlib.c

```diff
@@ -488,8 +488,7 @@ int nx_submit_job(nx_dde_t *src, nx_dde_t *dst, nx_gzip_crb_cpb_t *cmdp, nx_devp
 	if (ret) {
 		prt_err("%s:%d job did not complete in allotted time, ret %d\n",
 			__FUNCTION__, __LINE__, ret);
-		/* Panic! Don't know what to do */
-		exit(-1);
+		return -1;
```

Comment:
ETIME* are network return codes.  May be EAGAIN?  Your call better than mine
  https://man7.org/linux/man-pages/man3/errno.3.html

Reply:
The workbook mentions 240-249 and 251 error codes are `reserved for future system software use`. Would it be valid to create an internal CC error code for this case here?

Reply:
I think you will need to ask Haren and/or Bart.  It might be power architecture decision, may require RFC etc.  Haren stole one of the CC for the page faults.  He may know the procedure.

Reply:
I do agree with `-EAGAIN`, but that isn't a valid CC value according to the NX GZIP user manual.
I'm starting to re-think my original proposal because `nx_submit_job()` returns an `int`, while CC is `uint8_t`.
In other words, it's already not a proper CC, as opposed to what I said before.
What if we document the return value as the following?
```
nx_submit_job() returns 0 in case of success. In case of hardware or OS-detected error, returns a positive value as reported in the Completion Code from the CSB.  Returns -EAGAIN in case of timeout.
```